core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -297,6 +298,8 @@ protected synchronized InetSocketAddress getAddress() {
         return rpc.call();
       } catch (ThriftIOException e) {
         throw new IOException(e);
+      } catch (AlluxioTException e) {
+        throw Throwables.propagate(AlluxioException.from(e));
```

Comment:
We don't currently have any way to differentiate between recoverable and unrecoverable Alluxio exceptions, but I don't think we currently throw any which make sense to retry.

I've updated the javadoc to make it clear that we only retry for thrift-level exceptions.